app/models/social_networking/like.rb

```diff
@@ -8,15 +8,54 @@ class Like < ActiveRecord::Base
     # a Participant may only Like an item once
     validates :item_id, uniqueness: { scope: [:item_type, :participant_id] }
 
-    scope :for_today, lambda {
-      where(arel_table[:created_at]
-              .gteq(Date.today.beginning_of_day)
-              .and(arel_table[:created_at].lteq(Date.today.end_of_day)))
-    }
+    scope :for_today,
+          lambda {
+            where(arel_table[:created_at]
+                    .gteq(Date.today.beginning_of_day)
+                    .and(arel_table[:created_at].lteq(Date.today.end_of_day)))
+          }
 
-    scope :for_week, lambda {
-      where(arel_table[:created_at]
-              .gteq(Time.current.advance(days: -7).beginning_of_day))
-    }
+    scope :for_week,
+          lambda {
+            where(arel_table[:created_at]
+                    .gteq(Time.current.advance(days: -7).beginning_of_day))
+          }
+
+    def item_description
```

Comment:
This may be something to think about.  However, I am not sure this is what @ericcf was talking about and may have more to say:

```ruby
class Like < ActiveRecord::Base
  def shared_item_description
    shareable = Sharable.new(item)
    shareable.description
  end
end

module SocialNetworking
  class Shareable
    def initialize(item)
      @item = item
    end

    def description
      @item.try(:description) || "Unknown SharedItem Type (reporting), Item ID:#{@item.try(:id)}"
    end
  end
end

class Activity
  def description
    "Activity: #{self.activity_type.title}"
  end
end

module SocialNetworking
  class Profile
    def description
      "ProfileCreation: #{item.participant.display_name}"
    end
  end
end
```